shipyard-controller/handler/shipyardcontroller.go

```diff
@@ -537,7 +542,24 @@ func (sc *shipyardController) startTaskSequence(event models.Event) error {
 
 	taskSequence, err := sc.getTaskSequenceInStage(eventScope.Stage, taskSequenceName, shipyard)
 	if err != nil {
-		return err
+		msg := fmt.Sprintf("could not get definition of task sequence %s: %s", taskSequenceName, err.Error())
+		finishedEvent := event
+		finishedEventData := keptnv2.EventData{
+			Project: eventScope.Project,
+			Stage:   eventScope.Stage,
+			Service: eventScope.Service,
+			Labels:  eventScope.Labels,
+			Status:  keptnv2.StatusErrored,
+			Result:  keptnv2.ResultFailed,
+			Message: msg,
+		}
+		finishedEvent.Data = finishedEventData
+
+		sc.onSequenceFinished(finishedEvent)
+		return sc.sendTaskSequenceFinishedEvent(&models.EventScope{
+			EventData:    finishedEventData,
+			KeptnContext: event.Shkeptncontext,
+		}, taskSequenceName, event.ID)
```

Comment:
should also this code be pushed to its own function? This way it can be shared with the one in `onTriggerSequenceFailed()`

Reply:
you're right - that can be reused here as well

Reply:
Thank you :shipit: